core/src/main/java/kafka/server/SharePartitionManager.java

```diff
@@ -116,18 +117,23 @@ public CompletableFuture<Map<TopicIdPartition, PartitionData>> fetchMessages(Fet
         return future;
     }
 
-    public CompletableFuture<ShareAcknowledgeResponseData> acknowledge(ShareSession session, PartitionInfo partitionInfo) {
+    public CompletableFuture<ShareAcknowledgeResponseData> acknowledge(TopicIdPartition topicIdPartition,
+                                                                       List<ShareAcknowledgeRequestData.AcknowledgementBatch> ackBatches) {
```

Comment:
And why not to call with Map<TopicIdPartition,  List<ShareAcknowledgeRequestData.AcknowledgementBatch>>, else we have to iterate in KafkaApis on acknowledge. Rather we can do that in the acknowledge method here itself.

Reply:
Currently, the ShareAcknowledgeRequest lacks the GroupId. We should discuss whether that's an oversight or not (my feeling was that member ID is enough as a UUID, but I could be persuaded that GroupId would be valuable).

Reply:
Yeah I think we would require group id as share fetch do not explicitly say which offsets to fetch unlike consumer fetch hence an state per group is needed. I didn't bring this up earlier as thought it must be in request headers and I was wrong, also it makes sense to be not in headers anyways.

Reply:
Hi @apoorvmittal10 , agreed that we need groupId so I am adding it in the ShareAcknowledgeRequest.json and will be adding it in the code as well.

Reply:
> And why not to call with Map<TopicIdPartition, List<ShareAcknowledgeRequestData.AcknowledgementBatch>>, else we have to iterate in KafkaApis on acknowledge. Rather we can do that in the acknowledge method here itself.

Iteration on KafkaApis over Map<TopicIdPartition, List<ShareAcknowledgeRequestData.AcknowledgementBatch>> will have to happen in any case since we need to run various checks for `Errors.UNKNOWN_TOPIC_OR_PARTITION`, `Errors.TOPIC_AUTHORIZATION_FAILED` etc. Hence thought, we let only single record handling in this function. I am fine to change it if you disagree.

Reply:
This method signature makes sense to me now.

Reply:
Hmmm. may be good to start with and then we iterate on to make it better if needed.